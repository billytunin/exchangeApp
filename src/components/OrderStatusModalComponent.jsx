import React, { Component } from 'react';

class OrderStatusModalComponent extends Component {

  render() {
    return (
      <div className={`modal_wrapper ${this.props.modalData.show ? 'show' : 'hide'}`}>
        <div className="modal_container">
          <h2 className="title">{this.props.modalData.title}</h2>
          <div className={`body ${this.props.modalData.errorText === '' ? 'show' : 'hide'}`}>
            <p className="detalle">DETALLE:</p>
            <p className="buying caption">ORDEN PARA</p>
            <p className="buying value">{this.props.modalData.type}</p>
            <p className="monto caption">MONTO A {this.props.modalData.type}:</p>
            <p className="monto value">{this.props.modalData.amount}</p>
            <p className="precio caption">PRECIO POR CADA BTC:</p>
            <p className="precio value">{this.props.modalData.price}</p>

            <p className="notice">(AVISO: Para la finalidad de esta demo, las ordenes se aceptarán automaticamente y, por lo tanto, su balance se actualizará al instante)</p>
          </div>

          <p className={`errorText ${this.props.modalData.errorText === '' ? 'hide' : 'show'}`}>{this.props.modalData.errorText}</p>
          
          <button onClick={this.props.toggleModal}>OK</button>
          
        </div>
      </div>
    );
  }
}

export default OrderStatusModalComponent;
