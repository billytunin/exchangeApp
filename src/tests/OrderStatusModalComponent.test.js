import React from 'react';
import OrderStatusModalComponent from '../components/OrderStatusModalComponent.jsx';
import renderer from 'react-test-renderer';

let modalDataMock = {};

describe('Suite de tests del componente OrderStatusModalComponent', () => {
  beforeEach(() => {
    modalDataMock = {
      show: true,
      errorText: '',
      title: 'OrderStatusModalComponent test',
      type: 'comprar',
      amount: 1,
      price: 100000
    }
  })

  it('renderiza el componente sin errores y con las props iniciales', () => {
    const component = renderer.create(<OrderStatusModalComponent modalData={modalDataMock}></OrderStatusModalComponent>);
    const componentInstance = component.getInstance();

    expect(componentInstance.props.modalData.title).toEqual(modalDataMock.title);
    expect(componentInstance.props.modalData.type).toEqual(modalDataMock.type);
    expect(componentInstance.props.modalData.amount).toEqual(modalDataMock.amount);
    expect(componentInstance.props.modalData.price).toEqual(modalDataMock.price);
  });

  it('coincide con el Ãºltimo snapshot', () => {
    const component = renderer.create(<OrderStatusModalComponent modalData={modalDataMock}></OrderStatusModalComponent>);
    const JSONComponent = component.toJSON();

    expect(JSONComponent).toMatchSnapshot();
  });
})
