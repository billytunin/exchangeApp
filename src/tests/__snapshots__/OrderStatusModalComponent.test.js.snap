// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Suite de tests del componente OrderStatusModalComponent coincide con el último snapshot 1`] = `
<div
  className="modal_wrapper show"
>
  <div
    className="modal_container"
  >
    <h2
      className="title"
    >
      OrderStatusModalComponent test
    </h2>
    <div
      className="body show"
    >
      <p
        className="detalle"
      >
        DETALLE:
      </p>
      <p
        className="buying caption"
      >
        ORDEN PARA
      </p>
      <p
        className="buying value"
      >
        comprar
      </p>
      <p
        className="monto caption"
      >
        MONTO A 
        comprar
        :
      </p>
      <p
        className="monto value"
      >
        1
      </p>
      <p
        className="precio caption"
      >
        PRECIO POR CADA BTC:
      </p>
      <p
        className="precio value"
      >
        100000
      </p>
      <p
        className="notice"
      >
        (AVISO: Para la finalidad de esta demo, las ordenes se aceptarán automaticamente y, por lo tanto, su balance se actualizará al instante)
      </p>
    </div>
    <p
      className="errorText hide"
    >
      
    </p>
    <button
      onClick={undefined}
    >
      OK
    </button>
  </div>
</div>
`;
